File: task2.swift
import Foundation

print("строкa с кольцами: ", terminator: "")
let rings = readLine() ?? ""

/*
let input = try! String(contentsOfFile: "input.txt", encoding: .utf8).trimmingCharacters(in: .whitespacesAndNewlines)
let rings = input
let output = FileHandle(forWritingAtPath: "output.txt")!
*/

var sterchen = Array(repeating: Set<Character>(), count: 10)
let chars = Array(rings)

for i in stride(from: 0, to: chars.count, by: 2) {
    let color = chars[i]
    let sterchen_num = Int(String(chars[i+1]))!
    sterchen[sterchen_num].insert(color)
}

var result = [Int]()
for i in 0..<10 {
    if sterchen[i].contains("R") && sterchen[i].contains("G") && sterchen[i].contains("B") {
        result.append(i)
    }
}

let outputStr = "\(result.count)" + result.map { " \($0)" }.joined()
print(outputStr)

/*
output.write((outputStr + "\n").data(using: .utf8)!)
*/

File: task2.kt
import java.io.File

fun main() {
    print("Введите строку с кольцами: ")
    val rings = readLine() ?: ""

    /*
    val fin = File("input.txt").bufferedReader()
    val rings = fin.readLine()
    val fout = File("output.txt").printWriter()
    */

    val sterchen = Array(10) { mutableSetOf<Char>() }

    for (i in rings.indices step 2) {
        val color = rings[i]
        val sterchen_num = rings[i+1].digitToInt()
        sterchen[sterchen_num].add(color)
    }

    val result = (0..9).filter { sterchen[it].containsAll(setOf('R', 'G', 'B')) }

    println("${result.size}${result.joinToString(" ", " ")}")

    /*
    fout.println("${result.size}${result.joinToString(" ", " ")}")
    fin.close()
    fout.close()
    */
}
